### PR TITLE
add selection based on PV error (needed by HI @HLT)

### PR DESCRIPTION
**Pull request by mtosi**
this PR is meant to add the possibility of selecting tracks considering also the PV error
this feature was present in the AnalyticalTrackSelector, and it was missing in the new TrackCutClassifier

in last days, it turned out that it is used by HI

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by mtosi** [comment]
a new parameter has been added in the configuration of the TrackCutClassifier
but the corresponding declaration has been done by fillDescription

File: RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc [hunk 1]
```diff
@@ -93,6 +93,33 @@ namespace {
...
+  inline void drCut_wPVerror_par(reco::Track const & trk, int & nLayers, const float* par, const int * exp, Point const & bestVertexError, float drCut[]) {
+    float drE = trk.d0Error();
+    float rPVerr = sqrt(bestVertexError.x()*bestVertexError.y()); // shouldn't it be bestVertex.xError()*bestVertex.xError()+bestVertex.yError()*bestVertex.yError() ?!?!?
+
+    float drErrPV = std::sqrt(drE*drE+rPVerr*rPVerr);
+    for (int i=2; i>=0; --i) {
+      if (exp[i] != 0)
+	drCut[i] = par[i]*pow(nLayers,exp[i])*drErrPV;
+      else
+	drCut[i] = par[i]*drErrPV;
```
Comment (cvuosalo) [inline RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc]: The `if` statement would probably be better written as:
```
drCut[i] = par[i]*drErrPV;
if (exp[i] != 0)
	drCut[i] *= pow(nLayers,exp[i]);
```
Same for lines 102-105.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #13980 3236896e634e885c54341b663092d24f02667c52

Adding option for track selection to include the PV error, as needed by Heavy Ions. There should be no change in monitored quantities.

#13981 is the 80X version of this PR, and it was previously approved.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-04-14-2300 show no significant differences, as expected. The Jenkins comparisons are not complete, but since the Jenkins test results for #13981 show no problems, there is no cause for concern.

**Comment by davidlange6** [comment]
+1

**Pull request merged**